test/connectivity.jl

```diff
@@ -10,6 +10,12 @@ add_edge!(g,10,9)
 @test is_connected(HouseGraph())
 
 cc = connected_components(g)
+visitor = LightGraphs.TreeBFSVisitorVector(zeros(Int, nv(g)))
+label = zeros(Int, nv(g))
+label = LightGraphs.connected_components!(label, g)
```

Comment:
`connected_components!` produces a vector of labels where the label is the root of the BFS tree that found the vertex. This API is better for parallel access than the Vector{Vector{Int}} that the `connected_components` uses. We could wrap the labels version to support the api of the standard version. Then experts could use this directly.